bot/cogs/moderation/modlog.py

```diff
@@ -53,9 +61,10 @@ def __init__(self, bot: Bot):
                         'author': message.author.id,
                         'channel_id': message.channel.id,
                         'content': message.content,
-                        'embeds': [embed.to_dict() for embed in message.embeds]
+                        'embeds': [embed.to_dict() for embed in message.embeds],
+                        'attachments': attachment,
                     }
-                    for message in messages
+                    for message, attachment in zip(messages, attachments)
```

Comment:
This would need to be  `zip_longest` I believe.

Reply:
No, because `reupload_attachments()` return an empty list if there isn't any message attached

Reply:
Is that not exactly why it _would_ need to be `zip_longest`? If the attachments are empty then the resulting zip is empty too:

```py
>>> messages = [1, 2, 3]
>>> attachments = []
>>> list(zip(messages, attachments))
[]
```

Reply:
I think I misunderstood you. Did you mean if attachments are empty it means the messages would also be empty anyway?

Reply:
AFAIK `messages` will always be a list of `Message` and `attachments` a list of list (which could be empty) so I don’t think we need `zip_longest` since `messages` and `attachments` will always have the same length